src/ucs/vfs/sock/vfs_sock.c

```diff
@@ -25,25 +25,27 @@ typedef struct {
 } UCS_S_PACKED ucs_vfs_msg_t;
 
 
-void ucs_vfs_sock_get_address(struct sockaddr_un *un_addr)
+/* Ideally we should use ucs_fill_filename_template here, but ucs library is
+ * not linked by vfs_sock */
+static void ucs_vfs_sock_template(const char *tmpl, char *buf, size_t max)
 {
-    struct passwd *pw;
-    uid_t euid;
+    if (NULL != strstr(tmpl, "%i")) {
+        snprintf(buf, max, tmpl, (int)geteuid());
```

Comment:
what if %i appears more than once, or we have other instances of format sequence (%-something) in the string?

Reply:
> what if %i appears more than once, or we have other instances of format sequence (%-something) in the string?

well, then obviously it does not work.. Then we want to use UCS template function, and we can either:
- move it to vfs_sock module (then it means we need to move all other functions used by it)
- move impl to header (and also for all functions used inside it)
- link UCS with VFS daemon (and disable VFS_ENABLE to prevent spawning VFS fuse thread. Maybe also remove custom vfs_error logging)
What do you prefer?

Reply:
what if we include string.c directly in the ucx_vfs tool build?

Reply:
I evaluated each approach, and I think that the right way is to link UCS with VFS daemon.
In this case we can also get rid of code duplication in VFS.
I think we can just disable loading UCS modules, maybe with weak linkage

Reply:
> what if we include string.c directly in the ucx_vfs tool build?

No good..
Because that template also depends on many other functions (the most problematic one is `ucs_get_first_cpu`) which are defined in sys.c.

Reply:
ok let's try to link with UCS and see how it goes

Reply:
pushed
IMO it looks good, removed quite some code duplication

Reply:
If you like the approach, we can also get rid of separate lib `libucs_vfs_sock`: it should be part of UCS.
We can also maybe refactor `verbose` option, because now we can use regular UCX_LOG_LEVEL

Reply:
maybe in next PR, since the goal of this PR is customer bugfix so let's not delay it..